### PR TITLE
DEV-1829: Subaward Counts Initial Bugfixes

### PR DESCRIPTION
**Pull request by dpb-bah**
**Description:**
Fixes two bugs related to the discrepancies between similar API endpoints/downloads.

**Technical details:**
- Bug 1: When not filtering by any award type versus filtering by all the award types we know, there are a couple award types that have been included that the project owners declared shouldn't be used. Code changes simply remove them (could comment them out though it's skeptical if they would return).
- Bug 2: When finding parent awards for subawards, the equivalent ids included dashes in their `NONE`s causing a mismatch and unlinked subawards. This resolves that and allows more subawards to be linked.
- Clarifying subaward log

**Requirements for PR merge:**

1. [x] Unit & integration tests updated (N/A)
2. [x] API documentation updated (N/A)
3. [x] Necessary PR reviewers:
	- [x] Backend
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed (N/A)
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-1829](https://federal-spending-transparency.atlassian.net/browse/DEV-1829):
	- [x] Link to this Pull-Request
	- [x] Performance evaluation of affected (API | Script | Download)
	- [x] Before / After data comparison

Note: this doesn't complete said ticket, only a couple aspects discovered from said ticket.

**Area for explaining above N/A when needed:**
```
Unit & integration tests updated - no unit tests affected
API documentation updated - no documentation affected
Matview impact assessment complete - no matviews affected
Frontend impact assessment completed - no frontend affected
```

### CODE REVIEW

File: usaspending_api/broker/management/commands/load_fsrs.py [hunk 1]
```diff
@@ -66,13 +66,13 @@ def generate_unique_ids(row, award_type):
             # "CONT_AW_" + contract_agency_code + contract_idv_agency_code + contract_number + idv_reference_number
             return (
                 'CONT_AW_'
-                + (row['contract_agency_code'].replace('-', '') if row['contract_agency_code'] else '-NONE-')
+                + (row['contract_agency_code'].replace('-', '') if row['contract_agency_code'] else 'NONE')
```
Comment (tony-sappe) [inline usaspending_api/broker/management/commands/load_fsrs.py]: I'm confused by this, I see the dashes in the awards table. Example: `CONT_AW_97AS_-NONE-_07000198909DDLA70089C3135_-NONE-`

File: usaspending_api/awards/v2/lookups/lookups.py [hunk 2]
```diff
@@ -1134,11 +1134,6 @@
     'B': 'Purchase Order',
     'C': 'Delivery Order',
     'D': 'Definitive Contract',
-    'E': 'Unknown Type',
```
Comment (tony-sappe) [inline usaspending_api/awards/v2/lookups/lookups.py]: Perhaps leave as a detailed comment?

File: usaspending_api/broker/management/commands/load_fsrs.py (hunk 1 above)
Comment (dpb-bah) [reply usaspending_api/broker/management/commands/load_fsrs.py]: Weird. I thought it matched more with the `NONE`s instead of `-NONE-`s. Double checking.

Comment (dpb-bah) [reply usaspending_api/broker/management/commands/load_fsrs.py]: Figured it out. Check out line `104` in `get_award`:

```
            # Find the award to attach this sub-contract to, using the generated unique ID:
            award = (
                Award.objects.annotate(
                    modified_generated_unique_award_id=Func(
                        F('generated_unique_award_id'), Value('-'), Value(''), function='replace'
                    )
                )
                .filter(
                    modified_generated_unique_award_id=self.generate_unique_ids(row, award_type),
                    latest_transaction_id__isnull=False,
                )
                .distinct()
                .order_by("-date_signed")
                .first()
            )
```
It matches the two ids by removing the `-` and comparing them to this ID with the `-NONE-`, causing a mismatch,.

**Pull request merged**